server/command_string.go

```diff
@@ -80,6 +83,97 @@ func setCommand(c *Client) error {
 	return c.Resp("OK")
 }
 
+func setBitCommand(c *Client) error {
+	if len(c.args) != 3 {
+		return terror.ErrCmdParams
+	} else if c.args[1][0] == '-' {
+		return terror.ErrCmdParams
+	} else if (len(c.args[2]) != 1) || (c.args[2][0] != '0' && c.args[2][0] != '1') {
+		return terror.ErrCmdParams
+	}
+	
+        var (
+		i int
+                v   []byte
+		v_new []byte
+                err error
+		bitsCnt int
+		bitPos int
+		bytesCnt int
+        )
+
+	bitPos,err = strconv.Atoi(string(c.args[1]))
+	bitsCnt = bitPos+1
+	if err != nil {
+		return terror.ErrCmdParams
+	}
+
+	// offset starts with 0, we need to +1 to do calculation
+	if bitsCnt % 8 == 0 {
+		bytesCnt = bitsCnt / 8
+	} else {
+		bytesCnt = (bitsCnt / 8) + 1
+	}
+
+        v, err = c.tdb.Get(c.GetCurrentTxn(), c.args[0])
+        if err != nil {
+		// error
+		return err
+        } else if v == nil {
+		// the key is not exist yet, we should create it.
+		v = make([]byte, bytesCnt)
+
+		// init all the bits with zero
+		for i:=0;i<bytesCnt;i++ {
+			v[i] = 0
+		}
+
+		// set bit 0,1 operation
+		if c.args[2][0] == '0' {
+			v[bytesCnt-1]&=^(1<<(uint)(bitPos % 8))
+		} else if c.args[2][0] == '1' {
+			v[bytesCnt-1]|=(1<<(uint)(bitPos % 8))
+		}
+		
+		fmt.Println("bytesCnt is: %d", bytesCnt)
+	} else {
+		// get the key, then change its value
+		if bitsCnt <= len(v)*8  {
+		/// if set bit pos is less than or equal to it's length, just set it
+			// set bit 0,1 operation
+			if c.args[2][0] == '0' {
+				v[bytesCnt-1]&=^(1<<(uint)(bitPos % 8))
+			} else if c.args[2][0] == '1' {
+				v[bytesCnt-1]|=(1<<(uint)(bitPos % 8))
+			}
+		} else {
+		/// if set bit pos is bigger than it's length, append it and then chagne it
+			v_new = make([]byte, bytesCnt)
```

Comment:
if we can use `append` to append some bytes to the slice `v`?  we do not need to copy `v` to `v_new` manually.